File: examples/doublearray/doublearray.cpp
#include <iostream>

void doublearray(int array[], int size)
{
    for (int i = 0; i < size; i++)
    {
        int newvalue = array[i] * 2;
        array[i] = newvalue;
    }
}

int main()
{
    int nums[] = { 1, 2, 3 };
    int size = std::size(nums);
    doublearray(nums, size);
    
    for (int num : nums)
    {
        std::cout << num << std::endl;
    }
}

File: examples/doublearray/doublearray.java
public class Main
{
    static void doublearray(int[] array, int size)
    {
        for (int i = 0; i < size; i++)
        {
            int newvalue = array[i] * 2;
            array[i] = newvalue;
        }
    }
    
    public static void main(String[] args)
    {
        int[] nums = { 1, 2, 3 };
        int size = nums.length;
        doublearray(nums, size);
        
        for (int num : nums)
        {
            System.out.println(num);
        }
    }
}
